frontend/src/components/AdminDataScreen/AdminDataScreen.tsx

```diff
@@ -434,6 +905,277 @@ function AdminDataScreen() {
               ))}
             </div>
           )}
+
+          {entityTypeCard && (
+            <div className="logs">
+              <button
+                type="button"
+                onClick={() => {
+                  handleShowModal();
+                  setUpdate(false);
+                }}
+              >
+                <NewVolunteersCard isUpdate={isUpdate} />
+              </button>
+              {showModal && (
+                <FieldForm
+                  handleShow={handleShowModal}
+                  whichField="Entity Type" // choose between entitytype, foodtype, etc
+                  isUpdate={isUpdate}
+                  fieldName={currentEntityType} // the actual field, i.e. church, school, etc
+                />
+              )}
+              {search(q).map((item: any, index: any) => (
+                <FieldPage
+                  index={index}
+                  field={item}
+                  handleShow={handleShowModal}
+                  fieldHandler={entityTypeData}
+                />
+              ))}
+            </div>
+          )}
+          {/* TODO change the props of subsequent FieldForms to match above */}
+          {donorsFoodTypeCard && (
+            <div className="logs">
+              <button
+                type="button"
+                onClick={() => {
+                  handleShowModal();
+                  setUpdate(false);
+                }}
+              >
+                <NewVolunteersCard isUpdate={isUpdate} />
+              </button>
+              {showModal && (
+                <FieldForm
+                  handleShow={handleShowModal}
+                  whichField="Donors Food Type"
+                  isUpdate={isUpdate}
+                  vehicle={currentDonorsFoodType}
```

Comment:
make sure to change all of these to `fieldName`, (where it says vehicle) currently the field isn't populating inside the update form 
<img width="638" alt="Screenshot 2023-04-21 at 1 09 29 PM" src="https://user-images.githubusercontent.com/70536670/233725944-1bdf3e73-55fa-4dd9-879b-c2c4df9b2607.png">